src/hotspot/share/gc/shared/gcLocker.hpp

```diff
@@ -150,3 +151,16 @@ class GCLocker: public AllStatic {
 };
 
 #endif // SHARE_GC_SHARED_GCLOCKER_HPP
+
+/*
+ * GCLockerTimingDebugLogger tracks specific timing information for GC lock waits.
+ */
+class GCLockerTimingDebugLogger : public StackObj {
+private:
+  const char*  _log_message;
+  Ticks _start;
+
+public:
+  GCLockerTimingDebugLogger(const char* log_message);
+  ~GCLockerTimingDebugLogger();
+};
```

Comment:
There should be no code after the include guard on line 153. This class should be moved above it. With that said, this class is only used in gcLocker.cpp, so there's really no need to expose it through the gcLocker.hpp file, AFAICT.

Also, note that you are using `/* */` to add a comment about the class, but the rest of the code in this file uses `//`, so I'd prefer to see it changed.

Also note that GitHub complains that your addition lacks a newline at the end of the file. We recently went over the GC code base and fixed issues like that. Maybe there's a way to configure your editor to add one when making edits to the end of a file?

Reply:
Thanks for the feedback. Addressed in new PR revision.